spec/requests/annual_rent_updates_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+ RSpec.describe 'View annual rent updates API', type: :request do
+    let!(:fixed_data) { create_list(:fixed_datum, 10) }
+   
+    let(:rc_uplift_1) { fixed_data.first.rc_uplift }
+    let(:rc_uplift_2) { fixed_data.second.rc_uplift }
+    let(:rc_uplift_3) { fixed_data.third.rc_uplift }
+
+    let(:year_1) { Faker::Number.between(1, 10) + fixed_data.first.year }
+    let(:year_2) { fixed_data.second.year }
```

Comment:
I think here it is possible in theory for year_1 and year_3 to be the same as year_2, which would break the test `expect(json_response['no_of_accounts']).to eq(2)` on line 55. You need to make sure year_1 and year_3 are never the same as the year_2  (the year you want to pass to `get "/annual-rent-updates/#{year}"`. So maybe you could have something like this (if it's `year_2` that you want the annual updates for):

`let(:year_1) { Faker::Number.between(1, 10) + fixed_data.second.year }`
`let(:year_2) { fixed_data.second.year }`
`let(:year_3) { Faker::Number.between(11, 20) + fixed_data.second.year }`

Reply:
Makes sense.